### DISCUSSION TITLE
read an integer from serial port

### DISCUSSION THREAD

**Discussion by chongtianyifa** [bonsai-rx/bonsai#1160] [Q&A]
Hi, 
I need to read integer sent from another software. I configure the serial connection using CreateSerialPort and SerialStringRead. The problem is that the integer sent periodically by the other software via a virtual COM port  is not terminated by '\n' (no line separator, similar to print() in Arduino). Therefore, I cannot use \r\n to start a newline. What should I set here in my case?

Thanks,
Lei

**Comment by bruno-f-cruz** [comment]
I guess it depends how to do intend to parse your incoming data? Are you always expecting a fixed-length integer?

**Reply by chongtianyifa** [reply]
Hi,
Thank you for the help. My data is exclusively 1, a 8-bit byte.
I think it is not a issue of parsing incoming data as I got nothing after adding a Parse node right after the SerialStringRead node. In fact, I did not observe anything from the SerialStringRead node. It is very likely that bonsai did not know when to stop reading the data and feed it into the chain.
I try to read the message in the serial monitor of Arduino IDE, I got a square every time a byte comes in, whichever terminator I select (no line ending, NL, LR or NL&LR).
![image](https://user-images.githubusercontent.com/46367566/204032115-33d73d4c-00c7-46dc-b4e2-8a4ac2daca89.png)
![image](https://user-images.githubusercontent.com/46367566/204032149-9d1fa063-2860-493a-bc94-2bcd66fcb93f.png)

**Reply by bruno-f-cruz** [reply]
Isn't there any way you can add an additional CR character? From what I understand the serial communication currently implemented in bonsai operators relies on having a character to separate strings. I guess you could build your own parser and read one byte at a time? But it would likely be easier to just add the extra character is possible.

**Reply by chongtianyifa** [reply]
It seems that some future updates of Bonsai will allow it to read incoming bytes. Here is my post about this issue and Josh answered it. https://sanworks.io/forum/showthread.php?tid=3523&pid=4540#pid4540
